File: Sources/Indigox.UUM.HR/Model/HROrganizational.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Indigox.UUM.HR.Model
{
    public class HROrganizational
    {
        public string ID { get; set; }
        public string Name { get; set; }
        public string DisplayName { get; set; }
        public string ParentID { get; set; }
        public string ParentName { get; set; }
        public string Email { get; set; }
        public string Type { get; set; }
        public HRState State { get; set; }
        public bool Synchronized { get; set; }
        public DateTime ModifyTime { get; set; }
        public string Description { get; set; }
        public IDictionary<string, string> ExtendProperties { get; set; } = new Dictionary<string, string>();

        public IDictionary<string, string> CloneExtendProperties()
        {
            Dictionary<string, string> ret = new Dictionary<string, string>(this.ExtendProperties.Count);
            foreach (KeyValuePair<string, string> entry in this.ExtendProperties)
            {
                if (entry.Value != null)
                {
                    ret.Add(entry.Key, (string)entry.Value.Clone());
                }
                else
                {
                    ret.Add(entry.Key, "");
                }
            }
            return ret;
        }

    }
}

File: WebApp/res/js/Indigox/UUM/Application/HROrganizational.js
define("Indigox/UUM/Application/HROrganizational",
    [
        "Indigox.Web.JsLib.Models.RecordManager"
    ],
function (
    RecordManager
) {
    RecordManager.getInstance().register('HROrganizational', {
        columns: [
            { name: 'ID', text: '编号', type: String },
            { name: 'ParentID', text: '部门编号', type: String },
            { name: 'ParentName', text: '部门名称', type: String },
            { name: 'Name', text: '名称', type: String },
            { name: 'DisplayName', text: '显示名称', type: String },
            { name: 'Email', text: '邮件', type: String },
            { name: 'Type', text: '类型', type: String },
            { name: 'State', text: '状态', type: String },
            { name: 'Description', text: '描述', type: String },
            { name: 'Synchronized', text: '是否已同步', type: String }
        ],
        primaryKey: ['ID']
    });
});